base/strings/basic.jl

```diff
@@ -135,17 +135,44 @@ end
 
 ## Generic indexing functions ##
 
-prevind(s::DirectIndexString, i::Integer) = i-1
-prevind(s::AbstractArray   , i::Integer) = i-1
-nextind(s::DirectIndexString, i::Integer) = i+1
-nextind(s::AbstractArray   , i::Integer) = i+1
+prevind(s::DirectIndexString, i::Integer) = convert(Int, i)-1
+prevind(s::AbstractArray   , i::Integer) = convert(Int, i)-1
```

Comment:
I was hoping to do `::Int` on the function instead but this doesn't seem to work yet. Used `Int(x)-1` as a stopgap in the meantime.